cpp/src/arrow/compute/api_vector.h

```diff
@@ -188,6 +188,17 @@ class ARROW_EXPORT PartitionNthOptions : public FunctionOptions {
   NullPlacement null_placement;
 };
 
+/// \brief Options for cumulative sum function
+class ARROW_EXPORT CumulativeSumOptions : public FunctionOptions {
+ public:
+  explicit CumulativeSumOptions(std::shared_ptr<Scalar> start);
+  CumulativeSumOptions() : CumulativeSumOptions(std::make_shared<NullScalar>()) {}
```

Comment:
`nullptr` might be a better default?

Reply:
I thought `start` represented an index (default 0) and not an object?
Also, looking at the pandas cumsum API, would a skipna option be desirable? How are nulls handled at the moment?